### PR TITLE
Update to support ^1.0.0-rc.1

### PR DESCRIPTION
**Pull request by jwartofsky-yext**
Update to support ^1.0.0-rc.0

J=SLAP-2892
TEST=auto,manual

Auto tests pass.

Can run studio with npm run dev using
^1.0.0-rc.0 after modification.

Note that this change requires [this](https://github.com/yext/pages/commit/4cf67c99207883500b213e8aa68d9c6704c0210e) update in PagesJS, so will either need to update PagesJS locally or wait until the next PagesJS release.

### CODE REVIEW

**Comment by jwartofsky-yext** [comment]
This should not be merged until the next version of PagesJS is released to npm. This PR will also need to be updated to use the new PagesJS version.

**Comment by jwartofsky-yext** [comment]
Tested with dyext/pages-1.0.0-rc.1 and npm run dev. All is working as expected. This should be good to go.

File: e2e-tests/vite.config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { defineConfig } from "vite";
```
Comment (oshi97) [inline e2e-tests/vite.config.js:1]: remove this

File: apps/test-site/vite.config.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { defineConfig } from "vite";
```
Comment (oshi97) [inline apps/test-site/vite.config.js:1]: Is it possible for pagesjs to provide a default vite config if the user doesn't have one? I'd rather not have extra vite configs floating around the studio repo when we're just using the recommended default.

Comment (oshi97) [reply apps/test-site/vite.config.js:1]: If it's quick could we make the change in PagesJS? We don't need to release it right away but I'd like to remove these at a later date

File: package.json [hunk 3]
```diff
@@ -21,6 +21,8 @@
     "@typescript-eslint/parser": "^5.59.0",
     "@yext/eslint-config": "^1.0.0",
     "@yext/eslint-plugin-export-star": "^1.2.0",
+    "@yext/pages": "^1.0.0-rc.1",
```
[the remark this answers is not in the document]
Comment (jwartofsky-yext) [reply package.json:24]: I don't believe so. However this resolved the issue where we were pulling [this](https://www.npmjs.com/package/pages) package.

File: e2e-tests/tests/open-live-preview.spec.ts [hunk 4]
```diff
@@ -9,6 +9,6 @@ studioTest(
     await studioPage.livePreviewButton.click({ position: { x: 75, y: 0 } });
     const popup = await popupPromise;
     await popup.waitForLoadState();
-    await expect(popup).toHaveTitle("Pages Development Page");
+    await expect(popup).toHaveTitle("Pages Development");
```
Comment (jwartofsky-yext) [inline e2e-tests/tests/open-live-preview.spec.ts:12]: This change was needed to make the test pass

**Review by oshi97 — APPROVED** [review]

File: packages/studio-plugin/package.json [hunk 5]
```diff
@@ -33,7 +33,7 @@
     "@babel/preset-env": "^7.22.4",
     "@babel/preset-react": "^7.22.3",
     "@babel/preset-typescript": "^7.21.5",
-    "@yext/pages": "^1.0.0-beta.14",
+    "@yext/pages": "^1.0.0-rc.1",
```
Comment (nmanu1) [inline packages/studio-plugin/package.json:36]: I think the third party notices for `studio-plugin` need to be updated to reflect this change

**Pull request merged**